ui-admin/src/portal/siteContent/SiteContentVersionSelector.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useEffect, useId, useState } from 'react'
+import LoadingSpinner from 'util/LoadingSpinner'
+import Api, { SiteContent } from 'api/api'
+import Modal from 'react-bootstrap/Modal'
+import { instantToDefaultString } from 'util/timeUtils'
+import Select from 'react-select'
+
+type SiteContentVersionSelectorProps = {
+    portalShortcode: string
+    stableId: string
+    current: SiteContent
+    onDismiss: () => void
+    loadSiteContent: (stableId: string, version: number) => void
+}
+
+/** component for selecting versions of a survey */
+export default function SiteContentVersionSelector(props: SiteContentVersionSelectorProps) {
+  const { portalShortcode, stableId, current, onDismiss, loadSiteContent } = props
+  const [versionList, setVersionList] = useState<SiteContent[]>([])
+  const [isLoading, setIsLoading] = useState(true)
+  const [selectedVersion, setSelectedVersion] = useState<SiteContent>()
+  const selectId = useId()
+  const versionOpts = versionList.map(content => ({
+    label: <span>
+            Version <strong>{ content.version }</strong>
+      <span className="text-muted fst-italic ms-2">
+                ({instantToDefaultString(content.createdAt)})
+      </span>
```

Comment:
Definitely stealing this for the survey version selector. I had previously just shown version numbers but I like the context that the created date adds, and it'll be nice to have a consistent UX. I'll see if it makes sense to break this out into a re-usable component in my PR.